sycl/source/backend.cpp

```diff
@@ -102,7 +102,7 @@ __SYCL_EXPORT context make_context(ur_native_handle_t NativeHandle,
   Properties.stype = UR_STRUCTURE_TYPE_CONTEXT_NATIVE_PROPERTIES;
   Properties.isNativeHandleOwned = false;
   std::vector<ur_device_handle_t> DeviceHandles;
-  for (auto Dev : DeviceList) {
+  for (auto &Dev : DeviceList) {
```

Comment:
Can we also add const? 
```suggestion
  for (const auto &Dev : DeviceList) {
```

Reply:
That makes sense, thanks for the catch!